client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts

```diff
@@ -0,0 +1,120 @@
+import * as yup from 'yup';
+import Swal from 'sweetalert2';
+
+import placeTypesCodesHierarchy from 'Utils/placeTypesCodesHierarchy';
+import axios from 'Utils/axios';
+import useDBParser from "Utils/vendor/useDBParsing";
+import StoreStateType from 'redux/storeStateType';
+
+import InteractionEventDialogFields from '../InteractionsEventDialogContext/InteractionEventDialogFields';
+import InteractionEventContactFields from '../InteractionsEventDialogContext/InteractionEventContactFields';
+import InteractionEventDialogData from 'models/Contexts/InteractionEventDialogData';
+
+const useInteractionsForm = (props : Props): outCome => {  
+  const { interactionId, loadInteractions, closeNewDialog, closeEditDialog } = props;    
+  const { parseLocation } = useDBParser();
+    const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
+
+    const schema = yup.object().shape({
+        [InteractionEventDialogFields.PLACE_TYPE]: yup.string().nullable().required('סוג אתר חובה'),
+        [InteractionEventDialogFields.PLACE_SUB_TYPE]: yup.number().when(
+          InteractionEventDialogFields.PLACE_TYPE, {
+            is: placeType => placeType !== placeTypesCodesHierarchy.geriatric.code &&
+                             placeType !== placeTypesCodesHierarchy.office.code,
+            then: yup.number().required('תת סוג אתר חובה'),
+            otherwise: yup.number().nullable()
+          }
+        ),
+        [InteractionEventDialogFields.CONTACT_PERSON_PHONE_NUMBER]: yup.string().nullable().
+          matches(/^(0(?:[23489]|5[0-689]|7[2346789])(?![01])(\d{7}))|^$/, 'מספר טלפון לא תקין'),
+        [InteractionEventDialogFields.START_TIME]: yup.date().required(),
+        [InteractionEventDialogFields.END_TIME]: yup.date().required().when(
+          InteractionEventDialogFields.START_TIME, (startTime: Date) => {
+            return yup.date().min(startTime, 'שעה עד לא יכולה להיות קטנה משעה מ');
+        }),
+        [InteractionEventDialogFields.EXTERNALIZATION_APPROVAL]: yup.boolean().required('שדה חובה'),
+        [InteractionEventDialogFields.CONTACTS]: yup.array().of(yup.object().shape({
+            [InteractionEventContactFields.FIRST_NAME]: yup.string().nullable().required('שם פרטי חובה'),
+            [InteractionEventContactFields.LAST_NAME]: yup.string().nullable().required('שם משפחה חובה'),
+            [InteractionEventContactFields.PHONE_NUMBER]: yup.string().nullable().required('מספר טלפון חובה')
+              .matches(/^(0(?:[23489]|5[0-689]|7[2346789])(?![01])(\d{7}))$/, 'מספר טלפון לא תקין'),
+              [InteractionEventContactFields.ID]: yup.string().nullable().matches(/^\d+|^$/, 'ת.ז חייבת להכיל מספרים בלבד')
+              .length(9, 'ת.ז מכילה 9 מספרים בלבד')
+              .test('isValid', "ת.ז לא תקינה", (id: any) => {
+                let sum = 0;
+                if (id?.length === 9) {
+                  for (let i=0; i<9; i++) {
```

Comment:
why for and not forEach?

Reply:
forEach doesn't work for me

Reply:
You can use Array.from and then change the is to a string instead of any, adding a photo.
![image](https://user-images.githubusercontent.com/52863381/95017793-2a3bf200-0664-11eb-8521-73e3edd598f9.png)